hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientExceptions.java

```diff
@@ -209,6 +209,11 @@ public ClientExceptions(boolean jcacheAvailable) {
         register(ClientProtocolErrorCodes.STALE_APPEND_REQUEST_EXCEPTION, StaleAppendRequestException.class);
         register(ClientProtocolErrorCodes.NOT_LEADER_EXCEPTION, NotLeaderException.class);
         register(ClientProtocolErrorCodes.VERSION_MISMATCH_EXCEPTION, VersionMismatchException.class);
+        register(ClientProtocolErrorCodes.NO_SUCH_METHOD_ERROR, NoSuchMethodError.class);
```

Comment:
Actually, I have better refactoring in my head that I keep postponing. CleintExceptions and ClientExceptionFactory are two separate classes because we had a separate `client` module before. We can simply merge them so that there can not be a problem like this again. Let me work on that on this pr.

Reply:
Merging these classes in a separate PR would be awesome. I'm marking this one as resolved then.